src/main/java/io/confluent/idesidecar/restapi/connections/DirectConnectionState.java

```diff
@@ -72,14 +72,34 @@ public DirectConnectionState(
   }
 
   @Override
-  public ConnectionStatus getInitialStatus() {
+  protected ConnectionStatus getInitialStatus() {
     return new ConnectionStatus(
         null,
         spec.kafkaClusterConfig() != null ? KAFKA_INITIAL_STATUS : null,
         spec.schemaRegistryConfig() != null ? SR_INITIAL_STATUS : null
     );
   }
 
+  /**
+   * Check if the connection to the Kafka cluster is currently connected.
+   *
+   * @return true if the Kafka component is connected
+   */
+  public boolean isKafkaConnected() {
+    var status = getConnectionStatus();
+    return status.kafkaCluster() != null && status.kafkaCluster().isConnected();
+  }
+
+  /**
+   * Check if the connection to the Schema Registry is currently connected.
+   *
+   * @return true if the Kafka component is connected
+   */
+  public boolean isSchemaRegistryConnected() {
+    var status = getConnectionStatus();
+    return status.schemaRegistry() != null && status.schemaRegistry().isConnected();
+  }
+
```

Comment:
Convenience methods used at the moment in `RealDirectFetcher`, but adding these also makes it easier to override behavior in the `RealDirectFetcherTest` unit tests.